### PR TITLE
Add Foldable instance for Day

### PR DESCRIPTION
**Pull request by LiamGoodacre**
Also added `tupleDay` (`<⊗>`), for example:

```purescript
eg0 :: Array String
eg1 :: Maybe String
eg2 :: Map Int String

eg3 = (eg0 <⊗> eg1 <⊗> eg2) # foldMap
  \(v0 /\ v1 /\ v2) -> v0 <> v1 <> v2
```

Wasn't sure what to call the function and it's operator, feel free to rename.

### CODE REVIEW

File: src/Data/Functor/Day.purs [hunk 1]
```diff
@@ -117,3 +124,9 @@ instance comonadDay :: (Comonad f, Comonad g) => Comonad (Day f g) where
 
 instance comonadTrans :: Comonad f => ComonadTrans (Day f) where
   lower = runDay \get fx gy -> get (extract fx) <$> gy
+
+instance foldableDay :: (Foldable f, Foldable g) => Foldable (Day f g) where
```
Comment (paf31) [inline src/Data/Functor/Day.purs:128]: This seems like an unusual choice of instance to me. That `length (day _ fx gy)` is `length fx * length gy` seems a bit strange. I know `Foldable` has no laws to satisfy, but still. I noticed this instance also isn't present in `kan-extensions`, so I'd be interested to hear if there's a reason why.

File: src/Data/Functor/Day.purs [hunk 2]
```diff
@@ -94,6 +96,11 @@ assoclDay = runDay \phi f x -> runDay (\psi g h -> day identity (day (\a b c ->
 assocrDay :: forall f g h. (f ⊗ g) ⊗ h ~> f ⊗ (g ⊗ h)
 assocrDay = runDay \phi x h -> runDay (\psi f g -> day (#) f (day (\a b c -> phi (psi c a) b) g h)) x
 
+tupleDay :: forall f g x y . f x -> g y -> Day f g (Tuple x y)
```
Comment (paf31) [inline src/Data/Functor/Day.purs:99]: I feel like this maybe crosses the [Fairbairn threshold](https://mail.haskell.org/pipermail/libraries/2012-February/017548.html). I'm curious what you use it for?

File: src/Data/Functor/Day.purs (hunk 1 above)
Comment (LiamGoodacre) [reply src/Data/Functor/Day.purs:128]: My only real motivation for this was that it felt like it should have one, similar to how the Coyoneda instance works:
```purescript
toList (coyoneda t f) ≅ t <$> toList f
toList (day t f g) ≅ t <$> toList f <*> toList g
```

I'd also be interested to hear both: why it's not present in kan-extensions; and which alternative instances are possible.

File: src/Data/Functor/Day.purs (hunk 2 above)
Comment (LiamGoodacre) [reply src/Data/Functor/Day.purs:99]: This was mainly to facilitate the above example.  Will happily delete!

**Comment by LiamGoodacre** [comment]
Thanks for your feedback @paf31 :sparkles:

**Pull request closed without merging**